src/extrema.jl

```diff
@@ -0,0 +1,117 @@
+as peaks, and the remaining N entries control each of the N dimensions
+of `img`.
+
+# Citation:
+
+Lindeberg T (1998), "Feature Detection with Automatic Scale Selection",
+International Journal of Computer Vision, 30(2), 79–116.
+
+See also: [`BlobLoG`](@ref).
+"""
+function blob_LoG(img::AbstractArray{T,N}, σscales::Union{AbstractVector,Tuple};
+                  edges::Union{Bool,Tuple{Vararg{Bool}}}=(true, ntuple(d->false, Val(N))...), σshape=ntuple(d->1, Val(N))) where {T,N}
+    if edges isa Bool
+        edges = (edges, ntuple(d->edges,Val(N))...)
+    end
+    sigmas = sort(σscales)
+    img_LoG = Array{Float64}(undef, length(sigmas), size(img)...)
+    colons = ntuple(d->Colon(), Val(N))
+    @inbounds for isigma in eachindex(sigmas)
+        img_LoG[isigma,colons...] = (-sigmas[isigma]) * imfilter(img, Kernel.LoG(ntuple(i->sigmas[isigma]*σshape[i],Val(N))))
+    end
+    maxima = findlocalmaxima(img_LoG; dims=1:ndims(img_LoG), edges=edges)
+    [BlobLoG(CartesianIndex(tail(x.I)), sigmas[x[1]], img_LoG[x]) for x in maxima]
+end
+
+
+"""
+    findlocalmaxima(img; dims=coords_spatial(img), edges=true) -> Vector{CartesianIndex}
+
+Returns the coordinates of elements whose value is larger than all of
+their immediate neighbors.  `dims` is a list of dimensions to
+consider.  `edges` is a boolean specifying whether to include the
+first and last elements of each dimension, or a tuple-of-Bool
+specifying edge behavior for each dimension separately.
+"""
+findlocalmaxima(img::AbstractArray; dims=coords_spatial(img), edges=true) =
+        findlocalextrema(img, dims, edges, Base.Order.Forward)
+
+"""
+    findlocalminima(img; dims=coords_spatial(img), edges=true) -> Vector{CartesianIndex}
+
+Like [`findlocalmaxima`](@ref), but returns the coordinates of the smallest elements.
+"""
+findlocalminima(img::AbstractArray; dims=coords_spatial(img), edges=true) =
+    findlocalextrema(img, dims, edges, Base.Order.Reverse)
+
+
+findlocalextrema(img::AbstractArray{T,N}, dims, edges::Bool, order) where {T,N} = findlocalextrema(img, dims, ntuple(d->edges,Val(N)), order)
+
+function findlocalextrema(img::AbstractArray{T,N}, dims, edges::NTuple{N,Bool}, order::Base.Order.Ordering) where {T<:Union{Gray,Number},N}
+    dims ⊆ 1:ndims(img) || throw(ArgumentError("invalid dims"))
+    extrema = Array{CartesianIndex{N}}(undef, 0)
+    Iedge = CartesianIndex(map(!, edges))
+    R0 = CartesianIndices(img)
+    R = clippedinds(R0, Iedge)
+    I1 = _oneunit(first(R0))
+    Rinterior = clippedinds(R0, I1)
+    iregion = CartesianIndex(ntuple(d->d ∈ dims, Val(N)))
+    Rregion = CartesianIndices(map((f,l)->f:l,(-iregion).I, iregion.I))
+    z = zero(iregion)
+    for i in R
+        isextrema = true
+        img_i = img[i]
+        if i ∈ Rinterior
+            # If i is in the interior, we don't have to worry about i+j being out-of-bounds
+            for j in Rregion
+                j == z && continue
+                if !Base.Order.lt(order, img[i+j], img_i)
```

Comment:
I was actually thinking if it's better to expose this as argument `f` and make

```julia
findlocalextrema(f, img; [dims,] [edges])
```

Reply:
And furthermore, maybe it's better to have it work similar to `mapwindow`:

```julia
findall_window(f, img, window; border, indices)
```

Reply:
That's pretty interesting. From a pragmatic standpoint one would probably want to smooth with a separable filter and then use a 3x3 window rather than an nxn window (the latter is O(n^2), the former is O(n)). But I do see the point.